### DISCUSSION TITLE
Write codes for new exchange

### DISCUSSION THREAD

**Discussion by mertdogan** [Drakkar-Software/OctoBot#1692] [Q&A] [answered]
I want to write script codes for an exchange that listed in ccxt but not in your list of exchanges directory.

Where may i start and what must i do? Are there any step by step guide or your reccomendations for this?

**Comment by Herklos** [comment]
> I want to write script codes for an exchange that listed in ccxt but not in your list of exchanges directory.
> 
> Where may i start and what must i do? Are there any step by step guide or your reccomendations for this?

Hi @mertdogan, that's a good question. 
First we are listing all ccxt exchanges without any filter so it should appear in your OctoBot exchange list.
Secondly you can create a tentacle in tentacles/Trading/Exchange/your_exchange to create a python module that implement your exchange. You should have a look at other tentacle in tentacles/Trading/Exchange that implements fixes on ccxt implementations.

**Reply by mertdogan** [reply]
Thank you so much. I examine with files in tentacles\Trading\Exchange folder and i see a lot of different definitions and custom functions about each exchange.

What criteria do you take as basis when determining these definitions?

**Comment by GuillaumeDSM** [comment, marked as answer]
Hi @mertdogan,
These functions are local overrides of the [generic ccxt exchange](https://github.com/Drakkar-Software/OctoBot-Trading/blob/master/octobot_trading/exchanges/connectors/ccxt_exchange.py) that is used when the exchange is behaving exactly like ccxt is expecting it to.
Basically if an exchange has a different way of handling candle, open orders or anything in its API (different than on ccxt expected behavior), it's necessary to add a custom function for this part on the related exchange (in an exchange tentacle) so that the data given to OctoBot always has the same format

**Reply by mertdogan** [reply]
How can we understand and fix differense?

**Reply by GuillaumeDSM** [reply]
we need to manually test it and maintain it when there are changes

**Reply by mertdogan** [reply]
Thank you
